libs/dotnet-sdk/FeatureboardHttpClient.cs

```diff
@@ -0,0 +1,50 @@
+using Featureboard.DotnetSdk.Models;
+using Microsoft.Extensions.Logging;
+using System.Net;
+using System.Net.Http.Json;
+
+namespace Featureboard.DotnetSdk;
+
+internal class FeatureboardHttpClient : IFeatureboardHttpClient
+{
+  private readonly IHttpClientFactory _httpClientFactory;
+  private readonly ILogger<FeatureboardHttpClient> _logger;
+
+  public FeatureboardHttpClient(IHttpClientFactory httpClientFactory, ILogger<FeatureboardHttpClient> logger)
+  {
+    _httpClientFactory = httpClientFactory;
+    _logger = logger;
+  }
+
+  public async Task<(Dictionary<string, FeatureConfiguration>? features, string? lastModified)> FetchUpdates(string? lastModified, CancellationToken cancellationToken)
+  {
+    var request = new HttpRequestMessage(HttpMethod.Get, "all");
+    if (lastModified is not null)
+      request.Headers.Add("if-modified-since", lastModified);
+
+    var response = await _httpClientFactory.CreateClient(Constants.HttpClientName).SendAsync(request, cancellationToken);
+
+    if (response.StatusCode == HttpStatusCode.NotModified)
+    {
+      _logger.LogDebug("No changes");
+      return (null, lastModified);
+    }
+
+    if (response.IsSuccessStatusCode)
+    {
+      var features = await response.Content.ReadFromJsonAsync<Dictionary<string, FeatureConfiguration>>(cancellationToken: cancellationToken)
+                     ?? throw new ApplicationException("Unable to retrieve decode response content");
+
+      if (response.Headers.TryGetValues("last-modified", out var newLastModified))
+      {
+        lastModified = newLastModified.FirstOrDefault();
+      }
+
+      _logger.LogDebug("Fetching updates done, newLastModified={newLastModified}", lastModified);
+      return (features, lastModified);
+    }
+
+    _logger.LogError("Failed to get latest toggles: Service returned error ${status}(${statusText})", response.StatusCode, await response.Content.ReadAsStringAsync(cancellationToken));
```

Comment:
Prob isn't statusText, is probably responseBody. This is just a nit

Reply:
Fixed